Lib/fontmake/font_project.py

```diff
@@ -949,6 +951,12 @@ def run_from_designspace(
             preFilters, postFilters = loadFilters(designspace)
             filters = preFilters + postFilters
 
+        designspace_copy = self._load_designspace_sources(designspace)
+        source_fonts = [source.font for source in designspace_copy.sources]
+        if interp_outputs or any(COMPAT_CHECK_KEY in font.lib for font in source_fonts):
```

Comment:
should you simply check if COMPAT_CHECK_KEY is in the designspace.lib instead of any(font.lib)?

Reply:
That key doesn't get copied to the designspace.

Reply:
Yeah, in fact I'm suggesting that it should. Who's responsible for it? glyphsLib I imagine

Reply:
So, the Glyphs user sets that parameter on the GSFont or on the GSMaster? If it's on the font, then it's global and should belong to the DS.lib, if specific to indiviodual masters, then it's ok in font.lib but then should you not only include in the compat check the font masters that actually have the key? Currently you check all if any one has the key, IIUC.

Reply:
It's set on the font, so it should go to the DS lib. But we need the sources anyway to check them, so there's not much in it.

Reply:
I still think it would be nicer if we checked the DS.lib only, since this is a global parameter that applies to all masters, not some of them. The any() check followed by compat check for all is confusing to me.

Reply:
ok fine, one day we may fix glyphsLib to export global font parameters to the designspace.lib. For now I suggest you add a comment that says something like "glyphsLib currently stores this global GSFont parameter in all the master UFO's lib so we check if it's set in any of them"